TF SDK - round 2 (#2085)
saucelabs/sauce-docs: docs/test-fairy/sdk/map-locations.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

The TestFairy SDK does not require location permissions and does track location out of the box.

> In cases where developers wish to send location information to TestFairy, they will need to add location permissions to their app and use the code below to call TestFairy.updateLocation. After doing that, the location will be presented on a map as part of the session page.

Reviewer: Edit to use active voice.
```suggestion
If you want to send location information to TestFairy, add location permissions to your app and use the code below to call `TestFairy.updateLocation`. The location will then be presented on a map as part of the session page.
```

import android.content.Context;
import android.location.Criteria;
import android.location.Location;
import android.location.LocationListener;
import android.location.LocationManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.testfairy.TestFairy;

public class MainActivity extends AppCompatActivity implements LocationListener {

    private LocationManager locationManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Get the location manager
        locationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);

        Criteria criteria = new Criteria();
        String provider = locationManager.getBestProvider(criteria, false);
        Location location = locationManager.getLastKnownLocation(provider);

        // Initialize the location fields
        if (location != null) {
            onLocationChanged(location);
        }
    }

    @Override
    public void onLocationChanged(Location location) {
        TestFairy.updateLocation(location);
    }
}
```

</TabItem>

<TabItem value="ios">

> To record locations, all you need to do is call the static instance method `updateLocation` in the TestFairy class, passing in a collection of `CLLocations`.

Reviewer: ```suggestion
To record locations, call the static instance method `updateLocation` in the `TestFairy` class, passing in a collection of `CLLocations`.
```
